### PR TITLE
Update the Dialog (Modal) component

### PR DESCRIPTION
**Pull request by eldiegod**
fixes: #2312

### CODE REVIEW

File: packages/atlas/src/components/_overlays/Modal/Modal.tsx [hunk 1]
```diff
@@ -24,12 +26,12 @@ export const Modal: React.FC<ModalProps> = ({ children, show, className, noBoxSh
 
   return (
     <Portal containerRef={modalContainerRef}>
-      <CSSTransition in={show} timeout={200} classNames={transitions.names.fade} mountOnEnter unmountOnExit>
-        <ModalBackdrop />
+      <CSSTransition in={show} timeout={250} classNames={transitions.names.fade} mountOnEnter unmountOnExit>
```
Comment (drillprop) [inline packages/atlas/src/components/_overlays/Modal/Modal.tsx]: We should use `cVar` here: `timeout={parseInt(cVar('animationTimingMedium', true))}`

File: packages/atlas/src/components/_overlays/Modal/Modal.tsx [hunk 2]
```diff
@@ -24,12 +26,12 @@ export const Modal: React.FC<ModalProps> = ({ children, show, className, noBoxSh
...
+        <ModalBackdrop onClick={onExitClick} />
       </CSSTransition>
       <CSSTransition
         in={show}
-        timeout={200}
+        timeout={250}
```
Comment (drillprop) [inline packages/atlas/src/components/_overlays/Modal/Modal.tsx]: Same as above

File: packages/atlas/src/providers/confirmationModal.tsx [hunk 3]
```diff
@@ -57,15 +57,31 @@ export const useConfirmationModal = (modalProps?: DialogModalProps) => {
...
+          <DialogModal
+            {..._args}
+            onExitClick={
+              _args?.onExitClick
+                ? () => {
+                    _args?.onExitClick?.()
+                    _closeModal()
+                  }
+                : undefined
+            }
```
Comment (drillprop) [inline packages/atlas/src/providers/confirmationModal.tsx]: Hmm, I don't like how it's written, can we create the handler above component? You moved `args || modalProps` to variable anyway. I would also write this little differently i.e: 
```tsx
        const handleClick = () => {
          if (!_args?.onExitClick) {
            return undefined
          }
          _args?.onExitClick?.()
          _closeModal()
        }
```
This is almost the same and in my opinion more readable

File: packages/atlas/src/styles/transitions.ts [hunk 4]
```diff
@@ -18,7 +20,7 @@ export const transitions = {
     fadeAndSlide: 'fade-slide',
     slide: 'slide',
     slideDown: 'slide-down',
-    modal: 'modal',
+    modal: 'atlas-modal',
```
Comment (drillprop) [inline packages/atlas/src/styles/transitions.ts]: Why do we need to change that name? Do we have another "modal" somewhere?

**Review by drillprop — CHANGES_REQUESTED** [review]
These changes don't work well with the create member modal. This is how it looks on a 550px viewport

image

I'm also not sure if we should keep this particular dialog on the bottom of the screen. @toiletgranny wdyt?


PS: please add a link to the design in your PRs 🙏🏻

**Review by toiletgranny — CHANGES_REQUESTED** [review]
I'm so happy to see this coming alive so quickly! ❤️ I did, however, find a few issues that must be addressed.

1. **Modal animations.** I wasn't able to see any desktop-sized dialog appearing on screen with the scale animation. Not in the Storybook, nor in the app. Are you sure the animation was implemented properly? As a reminder, new transition animations are described [in Figma, here](https://www.figma.com/file/Pf31tuYpozYmpq163U2ho8/Web-components?node-id=6341%3A146413) (the "Animation" section at the very bottom).
2. **Dialog header.** — Please note, that there should be a `24px` (or `40px` starting with the SM breakpoint) of distance between the close button and the dialog title in the Header section. Currently, there's no gap in between these elements.
3. **Sign in** — This dialog has been left unaffected by our changes, it seems. Would it be a lot of work to have this component inherit the recent changes we made to dialogs and overlays, such as new transitions, new responsive behavior, rounded corners, the styling of the footer, etc.? At the very least, can we make sure it doesn't stop shrinking horizontally when reaching `690px` of width? Of course, ideally, all future updates to the styling/behavior of our dialogs/modals in Atlas are also picked up in this dialog. 
CleanShot 2022-04-12 at 15 53 16
4. **Create membership** — Same comment as above.
CleanShot 2022-04-12 at 15 38 14
5. **Processing and errors** — In the below two examples, both dialogs are — obviously — too wide. As a reminder, from the docs in Figma: "_The width of the modal contents should also be freely customizable, with 480px being a default value and 768px and 1120px being preferable presets for extending._"
CleanShot 2022-04-12 at 15 43 08
CleanShot 2022-04-12 at 15 51 30

---

Now, @kdembler, I wasn't able to access Studio in a way that would allow me to inspect dialog components over there. No matter which Polkadot account or chain I tried, I would always get stuck at the "Oops! An error occurred." screen. For this reason, I don't know if there are other issues with dialogs that are a result of these changes. Is there anything else I can try to access Studio in this PR's... _deployment_ (if that's the word 😅 )?

**Comment by eldiegod** [comment]
@toiletgranny did what I could on the existing dialogs I think most things should be fixed pls take a look again and lmk !

**Review by drillprop — CHANGES_REQUESTED** [review]
Still doesn't feel right to me:
image
image

image

@toiletgranny I don't see the answer to my question. I'm not sure if this topic was raised.  Maybe you discussed this with Diego somewhere else, so I'll ask again:
Should we keep the member dialog(only this particular dialog) at the bottom of the page on smaller screens? 👇🏻 

image
 
Shouldn't we leave this centered as it was before?

**Comment by toiletgranny** [comment]
Thanks, @eldiegod! I can confirm that the (1) Modal animations and (2) Dialog header issues I've listed in my previous comment are now fixed, thanks!

1. **Max-width** — As @drillprop pointed out, the issue of overly wide dialogs is still present, so for quick reference:
   - The max-width of both the [Transaction dialog](https://www.figma.com/file/Pf31tuYpozYmpq163U2ho8/Web-components?node-id=5207%3A204082) and [Alert dialog](https://www.figma.com/file/Pf31tuYpozYmpq163U2ho8/Web-components?node-id=6278%3A138198) components, starting on the SM breakpoint, should be both `480px`. In Figma, you'll find out that the max-width of the Transaction dialog component is set to `440px`, but this is due to historic reasons and I'm 100% fine with it following a global rule. 
   - The max-width of both the Sign in and Create membership dialogs should be set to `768px`.
   - For the future, let's keep in mind that `480px` should be the _default_ max-width value for all Dialog components. 
2. **Sign in**
   - The footer in this dialog def looks more aligned with our recent changes now, and the only thing that's still missing is the top border. 
   - Also, notice how the entire window is scrollable when there's not enough vertical space to display it fully in the viewport. Ideally, we're following the same pattern as with all other scrollable Dialog components, which is that only the body part is scrollable.  You can see this very clearly on the screen recording below when I'm switching between the Sign in (incorrect scrolling behavior) dialog and the Create membership dialog (correct scrolling behavior). 

https://user-images.githubusercontent.com/35307309/165274933-d3e97ac4-df90-4a1e-8e53-0f1da4d0543b.mp4

3. **Create membership** — It seems like either not all changes to Dialog component footers have been picked up in the Create membership dialog, or that we might have missed that one change entirely. Namely, the size of Button components inside Dialog footers has changed from Large to Medium. In the Create membership dialog, however, the size of the button in the future is still Large.

---

To answer your question, @drillprop: [following the new designs](https://www.figma.com/file/Pf31tuYpozYmpq163U2ho8/Web-components?node-id=6341%3A146413) ("Positioning, size, marings & overflow" section), all Dialog components (or, if you prefer, all components placed inside of the Modal component) should be placed at the bottom and be stretched to fill the entire width of the viewport. So **yes**, we should keep the member dialog (Sign in dialog) component at the bottom of the page on smaller screens, just like every other Dialog component up until the SM breakpoint.

**Comment by eldiegod** [comment]
> To answer your question, @drillprop: [following the new designs](https://www.figma.com/file/Pf31tuYpozYmpq163U2ho8/Web-components?node-id=6341%3A146413) ("Positioning, size, marings & overflow" section), all Dialog components (or, if you prefer, all components placed inside of the Modal component) should be placed at the bottom and be stretched to fill the entire width of the viewport. So **yes**, we should keep the member dialog (Sign in dialog) component at the bottom of the page on smaller screens, just like every other Dialog component up until the SM breakpoint.


@toiletgranny ok, I think I covered the bad widths and other stuff this time around, lmk if you find anything else!

**Review by drillprop — APPROVED** [review]
Looked at the code again and tested this visually and LGTM!  🎉

**Review by drillprop — CHANGES_REQUESTED** [review]
Oh sorry, there is still a warning in the console. This is happening when I'm opening create member modal
image

**Review by toiletgranny — APPROVED** [review]
Now it's perfect! Good to go from my perspective!

**Comment by eldiegod** [comment]
should be fixed

File: packages/atlas/src/components/_overlays/Dialog/Dialog.styles.ts [hunk 5]
```diff
@@ -81,25 +92,27 @@ const getDenseHeaderContentStyles = ({ denseHeader, noContentPadding }: ContentP
...
 export const footerDividersStyles = css`
-  box-shadow: inset 0 1px 0 0 ${oldColors.gray[600]};
+  box-shadow: ${cVar('effectDividersTop')};
   padding-top: var(--local-size-dialog-padding);
 `
 
 type FooterProps = {
-  hasAdditionalActions: boolean
-  additionalActionsNodeMobilePosition: 'bottom' | 'top'
+  'data-hasadditionalactions'?: boolean
```
Comment (drillprop) [inline packages/atlas/src/components/_overlays/Dialog/Dialog.styles.ts]:
```suggestion
  'data-has-additional-actions'?: boolean
```
This is not really readable

File: packages/atlas/src/components/_overlays/Dialog/Dialog.styles.ts [hunk 6]
```diff
@@ -108,12 +121,17 @@ export const Footer = styled.div<FooterProps>`
     flex-direction: row;
     align-items: center;
   }
+
+  &[data-hasadditionalactions='false'],
+  &:not([data-hasadditionalactions]) {
```
Comment (drillprop) [inline packages/atlas/src/components/_overlays/Dialog/Dialog.styles.ts]:
```suggestion
  &[data-has-additional-actions='false'],
  &:not([data-has-additional-actions]) {
```

File: packages/atlas/src/components/_overlays/Dialog/Dialog.tsx [hunk 7]
```diff
@@ -104,7 +114,7 @@ export const Dialog: React.FC<DialogProps> = ({
       {hasFooter && (
         <Footer
           dividers={dividers || actionDivider}
-          hasAdditionalActions={!!additionalActionsNode}
+          data-hasadditionalactions={!!additionalActionsNode}
```
Comment (drillprop) [inline packages/atlas/src/components/_overlays/Dialog/Dialog.tsx]:
```suggestion
          data-has-additional-actions={!!additionalActionsNode}
```

**Review by drillprop — CHANGES_REQUESTED** [review]

File: packages/atlas/src/components/_overlays/Dialog/Dialog.styles.ts (hunk 6 above)
Comment (eldiegod) [reply packages/atlas/src/components/_overlays/Dialog/Dialog.styles.ts]: yeah probably better

**Review by drillprop — APPROVED** [review]

**Pull request merged**